src/app/data-model-provider/Provider.h

```diff
@@ -47,6 +47,9 @@ class Provider : public ProviderMetadataTree
 public:
     virtual ~Provider() = default;
 
+    // Initialize the data model internal code to be ready to send and receive data model messages.
+    virtual void InitDataModel() {}
```

Comment:
Let's move this to startup - there seems no need to have a separate virtual call here.